lib/travis/build/appliances/setup_filter.rb

```diff
@@ -49,11 +49,14 @@ def owner_login
             fi
           ),
           redirect_io: %(
-            exec > >(
-              %{curl}
-              %{exports}
-              ruby ~/filter.rb %{args}
-            ) 2>&1
+            if [[ -z "$TRAVIS_FILTERED" ]]; then
```

Comment:
The pty strategy sets this variable (for unrelated reasons). This gets used in the two tests in `travis_terminate` to gauge whether it should be attempted to dismantle the filter. Not sure how defensiveness and the creation of a new env var relate.